client/src/components/__tests__/SignupModal.test.js

```diff
@@ -69,18 +73,64 @@ describe('input fields in form', () => {
 });
 
 describe('<SignupModal /> contains cancel and submit button', () => {
-  const toggleModal = jest.fn();
-  const wrapper = mount(<SignupModal show toggleModal={toggleModal} />);
+  const wrapper = mount(<SignupModal show closeModal={closeModal} />);
   const buttons = wrapper.find('button');
 
   test('should contain two buttons', () => {
     expect(buttons.length).toBe(2);
   });
 
-  test('should close modal by calling toggleModal on clicking cancel button', () => {
+  test('should close modal by calling closeModal on clicking cancel button', () => {
     const cancelBtn = wrapper.find('button[data-test="cancel"]');
     cancelBtn.simulate('click');
 
-    expect(toggleModal).toHaveBeenCalledWith('showSignupModal');
+    expect(closeModal).toHaveBeenCalled();
+  });
+});
+
+describe('<SignupModal /> form submission success', () => {
+  test('should call signupAPI if form data is correct', () => {
+    const wrapper = mount(<SignupModal show closeModal={closeModal} />);
+    const mockUser = {
+      username: 'nitin',
+      email: 'nitin@mail.com',
+      password: '123456',
+    };
+
+    signupAPI.mockImplementation(user => Promise.resolve(user));
+
+    const usernameInpt = wrapper.find('input[name="username"]');
+    const emailInpt = wrapper.find('input[name="email"]');
+    const passwordInpt = wrapper.find('input[name="password"]');
+    const repeatPasswordInpt = wrapper.find('input[name="repeat-password"]');
+
+    usernameInpt.simulate('change', { target: { value: mockUser.username } });
+    emailInpt.simulate('change', { target: { value: mockUser.email } });
+    passwordInpt.simulate('change', { target: { value: mockUser.password } });
+    repeatPasswordInpt.simulate('change', { target: { value: '123456' } });
+
+    wrapper.find('form').simulate('submit');
+
+    return Promise.resolve().then(() => {
+      expect(signupAPI).toHaveBeenCalledWith(mockUser);
+      expect(handleSuccess).toHaveBeenCalledWith(mockUser);
```

Comment:
Here done should be called if it is promise

Reply:
If we return a promise from test, then jest waits for the promise. No need to call done. 

> If your code uses promises, there is a simpler way to handle asynchronous tests. Just return a promise from your test, and Jest will wait for that promise to resolve. If the promise is rejected, the test will automatically fail.

https://jestjs.io/docs/en/asynchronous